lib/_http_server.js

```diff
@@ -1102,6 +1109,29 @@ function parserOnIncoming(server, socket, state, req, keepAlive) {
     });
   }
 
+  // Check if we should optimize empty requests (those without Content-Length or Transfer-Encoding headers)
+  const hasBodyHeaders = ('content-length' in req.headers) || ('transfer-encoding' in req.headers);
+  const isHeadOrGet = (req.method === 'HEAD' || req.method === 'GET');
+
+  // Apply optimization for all methods when optimizeEmptyRequests is enabled and there are no body headers
+  // For HEAD and GET methods, we optimize regardless of body headers for backward compatibility
+  const shouldOptimize = server[kOptimizeEmptyRequests] === true && (isHeadOrGet || !hasBodyHeaders);
+
+  if (shouldOptimize) {
+    // Fast processing where request "has" already emitted all lifecycle events.
+    // This avoids a lot of unnecessary overhead otherwise introduced by
+    // stream.Readable life cycle rules. The downside is that this will
+    // break some servers that read bodies for methods that don't have body headers.
+    req._dumped = true;
+    req._readableState.ended = true;
```

Comment:
Could we move the modifications of stream internals into stream files?
I wonder also that `endEmitted`/ `closeEmitted` are set here. Are they really emitted?

Reply:
> I wonder also that endEmitted/ closeEmitted are set here. Are they really emitted?

They are as if they were already emitted.